backend/internal/utils/file_util.go

```diff
@@ -24,6 +25,15 @@ func GetFileExtension(filename string) string {
 	return filename
 }
 
+// SplitFileName splits a full file name into name and extension.
+func SplitFileName(fullName string) (name, ext string) {
+	dot := strings.LastIndexByte(fullName, '.')
```

Comment:
LastIndex is safer in case the file name contains non-ASCII characters

```suggestion
	dot := strings.LastIndex(fullName, ".")
```